dashboard/app/models/hoc_census2017v3.rb

```diff
@@ -0,0 +1,39 @@
+# == Schema Information
+#
+# Table name: census_submissions
+#
+#  id                           :integer          not null, primary key
+#  type                         :string(255)      not null
+#  submitter_email_address      :string(255)
+#  submitter_name               :string(255)
+#  submitter_role               :string(255)
+#  school_year                  :integer          not null
+#  how_many_do_hoc              :string(255)
+#  how_many_after_school        :string(255)
+#  how_many_10_hours            :string(255)
+#  how_many_20_hours            :string(255)
+#  other_classes_under_20_hours :boolean
+#  topic_blocks                 :boolean
+#  topic_text                   :boolean
+#  topic_robots                 :boolean
+#  topic_internet               :boolean
+#  topic_security               :boolean
+#  topic_data                   :boolean
+#  topic_web_design             :boolean
+#  topic_game_design            :boolean
+#  topic_other                  :boolean
+#  topic_other_description      :string(255)
+#  topic_do_not_know            :boolean
+#  class_frequency              :string(255)
+#  tell_us_more                 :string(255)
+#  pledged                      :boolean
+#  created_at                   :datetime         not null
+#  updated_at                   :datetime         not null
+#
+# Indexes
+#
+#  index_census_submissions_on_school_year_and_id  (school_year,id)
+#
+
+class HocCensus2017v3 < HocCensus2017v2
```

Comment:
Again, there are a lot of different versions and types here. It would be nice to have some small description differentiating them.

Is there any similarity between the versioning between types, e.g. `Census2017v1` and `HocCensus2017v1` and their respective v2s, v3s, etc? If we want some common functionality between version numbers across types, it's probably worth using a model concern.

nit: I'd prefer more consistent naming, e.g. `Census<Type?><Year><Version>`, so we'd have `CensusHoc2017v1` instead of `HocCensus2017v1`.

On that note, it might be worth adding fields for type and year, like we did for the Pd::Applications (see https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/pd/application/application_base.rb#L43-L50). Not necessary since that's contained in the `type` but it might make organization and querying easier.

Reply:
There is no real mapping of versions for the different types. They just reflect changes made over time to the different forms. 

Good point on naming. I will change this. I had based the names on the FORMS table kinds (Census2017 and HocCensus2017) but I agree that starting all with "Census" would be an improvement.